app/services/xccdf_report_migration.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+# Migrates ref IDs from reports to comply with the expected standard.
+# Profile ref IDs should look like "xccdf_org.ssgproject.content_profile.*"
+# Rule ref IDs should look like "xccdf_org.ssgproject.content_rule.*"
+class XCCDFReportMigration
+  def initialize(account, noop)
+    @account = account
+    @noop = noop
+  end
+
+  def logger
+    Rails.logger
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def run
+    xccdf_profiles = broken_profiles
+    logger.info "Found #{xccdf_profiles.count} profiles from XCCDF reports."\
+      ' Renaming profiles...'
+    rename_profiles(xccdf_profiles)
+    logger.info "Profiles rename finished \n-------------------------------\n"\
+      'Some profiles might have not been renamed if the name '\
+      'conflicts with another profile in the same account.'\
+      "Let's proceed to migrating the XCCDF profile results into the"\
+      ' DS profile results'
+    xccdf_profiles = broken_profiles
+    logger.info "Found #{xccdf_profiles.count} profiles from XCCDF reports."
+    migrate_profiles(xccdf_profiles)
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  def broken_profiles
+    Profile.where(account_id: @account.id).reject do |profile|
+      profile.ref_id.match?(/xccdf_org.ssgproject.content_profile_.*/)
+    end
+  end
+
+  def rename_profiles(xccdf_profiles)
+    return if @noop
+
+    xccdf_profiles.map do |profile|
+      profile.update(
+        ref_id: "xccdf_org.ssgproject.content_profile_#{profile.ref_id}"
+      )
+      migrate_rules(profile.rules, profile) if profile.errors.empty?
+    end
+  end
+
+  def find_matching_profile(xccdf_profile)
+    Profile.where(
+      account_id: @account.id,
+      ref_id: "xccdf_org.ssgproject.content_profile_#{xccdf_profile.ref_id}"
+    ).first
+  end
+
+  def migrate_profile_hosts(xccdf_profile, ds_profile)
+    return if @noop
+
+    # Justification: Enhance the speed of the mass-migration
+    # rubocop:disable Rails/SkipsModelValidations
+    logger.info ProfileHost.where(profile_id: xccdf_profile.id)
+                           .update_all(profile_id: ds_profile.id)
```

Comment:
This can fail if another ProfileHost exists already with this host/profile combo.

Reply:
Fixed